backend/app/ingestion/models.py

```diff
@@ -163,3 +164,23 @@ async def get_similar_n_chunks(
         )
 
     return results_dict
+
+
+async def rerank_chunks(
+    similar_chunks: dict[int, DocumentChunk],
+    query_text: str,
+    n_top_rerank: int,
+    request: Request,
+) -> dict[int, DocumentChunk]:
+    """
+    This function reranks the chunks using the cross-encoder
+    """
+    encoder = request.app.state.crossencoder
+    contents = similar_chunks.values()
+    scores = encoder.predict([(query_text, content.text) for content in contents])
```

Comment:
@sidravi1 - the first time I tested this (and the cross-encoder is downloaded/ used) I get this warning.

```
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
        - Avoid using `tokenizers` before the fork if possible
        - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
```

Reply:
I know this is minor (has popped up in lots of other projects and never caused an issue!) but would be nice to resolve.

Reply:
Yeah... i couldn't figure out the right fix.
Easy thing to do is `TOKENIZERS_PARALLELISM=false` but not sure what the repercussions of that are. I'll raise a separate ticket so we can track it